packages/better-auth/src/plugins/api-key/index.ts

```diff
@@ -165,21 +167,15 @@ export const apiKey = (options?: ApiKeyOptions) => {
 							? key
 							: await defaultKeyHasher(key);
 
-						const apiKey = await ctx.context.adapter.findOne<ApiKey>({
-							model: API_KEY_TABLE_NAME,
-							where: [
-								{
-									field: "key",
-									value: hashed,
-								},
-							],
+						const apiKey = await validateApiKey({
+							hashedKey: hashed,
+							ctx,
+							opts,
+							schema,
 						});
 
-						if (!apiKey) {
-							throw new APIError("UNAUTHORIZED", {
-								message: ERROR_CODES.INVALID_API_KEY,
-							});
-						}
+						await routes._deleteAllExpiredApiKeys(ctx.context);
```

Comment:
just use `await deleteAllExpiredApiKeys(ctx.content)`. I don't see a reason for duplicating the function in this case tbh

Reply:
@rxri the endpoint and the middleware take a different context, that's why i made them separate. Also, I think it's better to keep them isolated, because I don't need the whole route, just the method which is shared all across.
<img width="1894" alt="image" src="https://github.com/user-attachments/assets/0db96c66-7857-4f7d-a363-e33d98cc5922" />